### DISCUSSION TITLE
Timestamps, and refresh_status

### DISCUSSION THREAD

**Discussion by BillyCroan** [ianjwhite99/sync-connect-sdk#57] [Q&A] [answered]
timestamps returned with statuses are UTC right?

And are they meant to represent the 'as of' time, because the car is not 'pinged' in realtime when querying the ford cloud?

I know that in the fordpass app I can swipe down and it will ping out to the car and wait and get a message back and update what the app shows, so there has to be some way to summon updated info from the car.  does .refresh_status do that?

Anyone know what the limits are to refresh_status, or why I wouldn't set up a cron job to refresh status every 5 minutes while ignition is on, every hour while off?

**Comment by HKLM** [comment, marked as answer]
[Based of my answer](https://github.com/ianjwhite99/connected-car-python-sdk/discussions/31#discussioncomment-1856463) to a similar question. 

when you send the "status" command, you are "asking the Ford API for the status". The Ford API server responds with the info, that is in the servers record (data from the last time the vehicle sent a update i.e. turned the vehicle off) This is not the same as "asking the vehicle" for the status,

To get the Ford API server to update the info it has on the vehicle. 
1. By sending the "refresh_status" command, you can manually force the Ford API to tell the vehicle, the server wants a full refresh of its status.  The vehicle then tells the Ford API it's new current status (as of right now time),  

2. At this time you must wait for the above (refresh_status) command to complete. 

3. Remember its the Ford API severs that has this new data. **You do not** have that info. Now you need to send a "status" command and the Ford API will respond with that new data it last got.


I dont think sending refresh_status commands every 5 minutes is a good idea. You would quickly use up alot of the Ford API servers time, that in return, would get attention of the server admin. And they will do something to make it stop or to limit the requests, etc so that things get back to normal.  Remember what we are doing here is not approved by ford. They do not have to allow us anything. They have changed things in the past I do believe to stop unauthorized apps/people from accessing the APIs.  Ford has been kind and offers this as a courtesy. It costs them money to use AT&T's network to get the data from the vehicles.   Overwhelm the system and Ford would likely to become a pay-for-use system, Use it responsibly.
